lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -1745,6 +2051,23 @@ impl FileBothDirectoryInformation {
         w.extend_from_slice(&util::to_unicode(&self.file_name, false));
         Ok(w)
     }
+
+    fn from(fso: FileSystemObject) -> RdpResult<Self> {
+        let file_attributes = if fso.file_type == FileType::Directory {
+            flags::FileAttributes::FILE_ATTRIBUTE_DIRECTORY
+        } else {
+            flags::FileAttributes::FILE_ATTRIBUTE_NORMAL
+        };
+        Ok(FileBothDirectoryInformation::new(
+            i64::try_from(fso.last_modified)?,
```

Comment:
Since we are going to re-use the same variable maybe it would be better to move that `i64:try_from(fso.last_modified)?` out of the initialization call  into separate variable and then use it rather than trying to cast it four times?